### PR TITLE
feat: dev-docs out of office replacement

### PR DESCRIPTION
**Pull request by hamza221**
### ☑️ Resolves

*part of https://github.com/nextcloud/server/issues/40398

### 🖼️ Screenshots
image
image
image
image

### CODE REVIEW

File: developer_manual/client_apis/OCS/ocs-out-of-office-api.rst [hunk 1]
```diff
@@ -27,21 +27,25 @@ Fetch data of the ongoing out-of-office period of a user.
...
++---------------------------------+--------+---------------------------------------------------------------------+
+| ``firstDay``                    | string | First day of the absence in format ``YYYY-MM-DD``                   |
++---------------------------------+--------+---------------------------------------------------------------------+
+| ``lastDay``                     | string | Last day of the absence in format ``YYYY-MM-DD``                    |
++---------------------------------+--------+---------------------------------------------------------------------+
+| ``status``                      | string | Short text that is set as user status during the absence            |
++---------------------------------+--------+---------------------------------------------------------------------+
+| ``message``                     | string | Longer multiline message that is shown to others during the absence |
++---------------------------------+--------+---------------------------------------------------------------------+
+| ``replacementUserId``           | string | ID of the replacement user                                          |
```
Comment (ChristophWurst) [inline developer_manual/client_apis/OCS/ocs-out-of-office-api.rst:45]: can we express that it's nullable?

File: developer_manual/client_apis/OCS/ocs-out-of-office-api.rst [hunk 2]
```diff
@@ -27,21 +27,25 @@ Fetch data of the ongoing out-of-office period of a user.
...
++---------------------------------+--------+---------------------------------------------------------------------+
+| ``replacementUserDisplayName``  | string | Displayname of the replacement user                                 |
```
Comment (ChristophWurst) [inline developer_manual/client_apis/OCS/ocs-out-of-office-api.rst:47]: same

**Review by ChristophWurst — COMMENTED** [review]
Did not catch this in the server repo but let's use null for no replacement set instead of an empty string

File: developer_manual/client_apis/OCS/ocs-out-of-office-api.rst (hunk 1 above)
Comment (st3iny) [reply developer_manual/client_apis/OCS/ocs-out-of-office-api.rst:45]: Maybe `string|null`

**Review by ChristophWurst — APPROVED** [review]

**Pull request merged**